### PR TITLE
[SVA-308] Add exclude filter for newsItemsDataProviders query

### PR DESCRIPTION
**Pull request by tonekk**
Added possibility to filter `DataProvider` in `newsItemDataProviders` query with `exclude` parameter

SVA-308

**New query example:**

```graphql
query NewsItems($ids: [ID!]) {
  newsItems(excludeDataProviderIds: $ids) {
    id
    contentBlocks {
      title
    }
    dataProvider {
      id
    }
  }
}
```

Example variables:
```graphql
{
  "ids": [
    1,
    3,
    5
  ]
}
```

### CODE REVIEW

File: app/graphql/types/query_type.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@ def waste_location_type(id:)
     def news_items_data_providers(category_id: nil)
       return DataProvider.joins(:news_items).order(:name).uniq if category_id.blank?
 
-      NewsItem.with_category(category_id).map(&:data_provider).sort_by do |data_provider|
+      NewsItem.with_category(category_id).includes(:data_provider).map(&:data_provider).sort_by do |data_provider|
```
Comment (donni106) [inline app/graphql/types/query_type.rb:121]: do you mind line breakings here to satisfy linter?

Bildschirmfoto 2022-01-25 um 13 00 57

File: app/graphql/resolvers/news_items_search.rb [hunk 2]
```diff
@@ -50,7 +51,11 @@ def apply_data_provider(scope, value)
   end
 
   def apply_data_provider_id(scope, value)
-    scope.joins(:data_provider).where(data_providers: { id: value })
+    scope.where(data_provider_id: value)
+  end
+
+  def apply_exclude_data_provider_ids(scope, value)
+    scope.exclude_data_providers(value)
```
Comment (donni106) [inline app/graphql/resolvers/news_items_search.rb:58]: why is the one `where` for a single id above here in this file and this for multiple ids (`where.not(data_provider_id: data_provider_ids)`) placed in the model?

File: app/graphql/types/query_type.rb (hunk 1 above)
Comment (tonekk) [reply app/graphql/types/query_type.rb]: Solved in https://github.com/ikuseiGmbH/smart-village-app-mainserver/commit/444c7559fc08f7ab7a3cb7d777551753d812ea5b

File: app/graphql/resolvers/news_items_search.rb (hunk 2 above)
Comment (tonekk) [reply app/graphql/resolvers/news_items_search.rb]: Solved in https://github.com/ikuseiGmbH/smart-village-app-mainserver/commit/444c7559fc08f7ab7a3cb7d777551753d812ea5b

**Review by donni106 — APPROVED** [review]

**Pull request merged**